kernel/Base/PhysicalFace.h

```diff
@@ -362,6 +363,24 @@ class PhysicalFace final {
     void setQuadraturePointIndex(std::size_t index);
 
    private:
+    void resetLazyCaches(std::size_t currentUnknowns);
+    void computeBasisFunctionDeriv(
+        std::vector<LinearAlgebra::SmallVector<DIM>>& values,
+        std::size_t unknown);
+    void computeBasisFunctionCurl(
+        std::vector<LinearAlgebra::SmallVector<DIM>>& values,
+        std::size_t unknown);
+    void computeBasisFunctionDiv(std::vector<double>& values,
+                                 std::size_t unknown);
```

Comment:
We cannot use the return type of these functions to return values, because of allocations? Would make the signature nicer.

Reply:
That is indeed the idea. This is called for every quadrature point on every face. Each triangular face has say 4 quadrature points and a mesh can easily have several thousands of faces. The only upside is that the vector usually stays the same size (that is why there is a split internal and boundary faces in `PhysicalFace`).